packages/vscode-extension/src/webview/providers/RoutesProvider.tsx

```diff
@@ -0,0 +1,55 @@
+import { createContext, useContext, useEffect, useState } from "react";
+import { useProject } from "./ProjectProvider";
+
+declare global {
+  interface Window {
+    RNIDE_lastRouteList?: Route[];
+  }
+}
+
+const RoutesContext = createContext<Route[] | null>(null);
+
+export type Route = {
+  path: string;
+  filePath: string;
+  children: Route[];
+  dynamic: { name: string; deep: boolean; notFound?: boolean } | null;
+  type: string;
+};
+
+export default function RoutesProvider({ children }: { children: React.ReactNode }) {
+  const [routes, setRoutes] = useState<Route[] | null>(null);
+  const { project } = useProject();
+
+  useEffect(() => {
+    if (window.RNIDE_lastRouteList) {
+      setRoutes(window.RNIDE_lastRouteList);
+    }
```

Comment:
I think it'd be best to wait for changes from #1154 – there we save routes history in state and we should do the same with the available routes list avoiding this type of hacks